typescript-function/src/function.ts

```diff
@@ -0,0 +1,18 @@
+import { BinarisFunction } from './binaris';
+
+interface DataWithName { name: string; }
+
+function bodyHasName(body: unknown): body is DataWithName {
```

Comment:
why so complex?
I think the following is simpler, does not use coercion nor immutable variables.
```
function getName(body: unknown, ctx: FunctionContext): string {
  if (typeof ctx.request.query.name === 'string') {
    return ctx.request.query.name;
  }
  if (typeof body === 'object' && body && typeof body['name'] === 'string') {
    return body['name'];
  }
  return 'World';
}

export const handler: BinarisFunction = async (body, ctx): Promise<string> => {
  const name = getName(body, ctx);
  return `Hello ${name}!`;
};
```

Reply:
I have to think about this. I'm not sure I entirely agree. There is more to consider than just writing the most concise code.

Reply:
Note that typescript compiler does not accept it in strict mode (E: Element implicitly has an 'any' type because type '{}' has no index signature.)

The `unknown` type almost forces coercion - perhaps it is too strict and we should publish the interface with `any`.

Reply:
100% agree. I think the existing code (my original + vlads revisions) is the best solution while we have `body` as `unknown`. If we change that component, your solution makes a lot of sense.